### PR TITLE
refactor: on-chain DID type

### PR DESCRIPTION
**Pull request by wischli**
## fixes KILTProtocol/ticket#1116
Applies changes to DID type on the mashnet-node proposed in https://github.com/KILTprotocol/mashnet-node/pull/118 as a result of mashnet-node CI failure due to clippy.

## How to test:

```
docker pull kiltprotocol/mashnet-node:develop
docker run -p 9944:9944 kiltprotocol/mashnet-node:develop --dev --ws-port 9944 --ws-external --rpc-external
yarn test:integration
```

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by wischli** [comment]
CI will fail until the corresponding [mashnet-node PR](https://github.com/KILTprotocol/mashnet-node/pull/118) is merged.

**Comment by wischli** [comment]
Have to wait for new `mashnet-node:develop` image to be published before expecting success in integration tests..

File: packages/core/src/did/Did.utils.ts [hunk 1]
```diff
@@ -143,7 +141,7 @@ export function verifyDidDocumentSignature(
   if (identifier !== id) {
     throw SDKErrors.ERROR_DID_IDENTIFIER_MISMATCH(identifier, id)
   }
-  const unsignedDidDocument = { ...didDocument }
+  const unsignedDidDocument = { ...didDocument, signature: undefined }
```
Comment (tjwelde) [inline packages/core/src/did/Did.utils.ts:144]: Why is that necessary, when we remove the signature right after?

Comment (wischli) [reply packages/core/src/did/Did.utils.ts:144]: The linter complained about the deletion.

Comment (wischli) [reply packages/core/src/did/Did.utils.ts:144]: I am happy to remove this as this is independent of the PR.

Comment (rflechtner) [reply packages/core/src/did/Did.utils.ts:144]: proposal: instead of deletion below, why don't we do 
```typescript
const { signature, ...unsignedDidDocument } = didDocument
```

Comment (wischli) [reply packages/core/src/did/Did.utils.ts:144]: Not sure whether this is fine because then the `signature` key still exists.

Comment (rflechtner) [reply packages/core/src/did/Did.utils.ts:144]: it does not in unsignedDidDocument, but of course we then have a variable in scope with that name - not sure how to improve this (the linter will complain)

Comment (wischli) [reply packages/core/src/did/Did.utils.ts:144]: Okay so I will change it to this, which won't have a linter complain:
```ts
  const { signature, ...unsignedDidDocument } = didDocument
  return Crypto.verify(
    Crypto.hashObjectAsStr(unsignedDidDocument),
    signature,
    getAddressFromIdentifier(identifier)
  )
```

Comment (wischli) [reply packages/core/src/did/Did.utils.ts]: Fixed in 692eca0

Comment (rflechtner) [reply packages/core/src/did/Did.utils.ts]: oooh I love it! didn't see we need the signature anyway, that's perfect.

File: packages/core/src/did/Did.utils.ts [hunk 2]
```diff
@@ -21,25 +20,24 @@ import {
...
+  if (encoded.isSome) {
+    const did = encoded.unwrap()
     return {
       identifier,
-      publicSigningKey: decoded[0],
-      publicBoxKey: decoded[1],
-      documentStore,
+      publicSigningKey: did.signKey.toString(),
+      publicBoxKey: did.boxKey.toString(),
+      documentStore: did.docRef.unwrapOrDefault().toHuman()?.toString(),
```
Comment (rflechtner) [inline packages/core/src/did/Did.utils.ts:40]: doesn't this just build an string presentation of an array? why do we no longer use `hexToString`

Comment (wischli) [reply packages/core/src/did/Did.utils.ts]: Good point, fixed in f494360

Comment (rflechtner) [reply packages/core/src/did/Did.utils.ts]: I think to get to the target of `string | undefined` we even need to do something like 
```typescript
did.docRef.isSome ? hexToString(did.docRef.toHex()) : undefined
```
(sidetone: seems like `toHex()` handles a `none` option directly by returning `0x`)

Comment (wischli) [reply packages/core/src/did/Did.utils.ts]: Fixed in a8496ed. I was wondering why we don't to any real integration tests with DIDs or was I not looking thoroughly enough? I wanted to test my and your proposal but did not want to invest time into setting up a test for this.

**Review by rflechtner — APPROVED** [review]
LGTM
